### ISSUE TITLE
Android App

### ISSUE THREAD

**Issue by drusepth** [indentlabs/notebook#84]
I'm not sure people will want to write a ton of freeform text on mobile, but at the very least we should offer an app that caches their content offline and provides an easy way to refer to any content and/or add details to anything. 

Should probably do the same online/offline syncing we do in #82 as well.

**Comment by JamesPierce82** [comment]
I see this issue was opened about 4 months ago, any news on starting an app for this?

**Comment by Cantido** [comment]
We'd love to do it, it's just that neither @drusepth nor I are familiar with the process. Java is my main language so I'd to pick it up, but right now it's the best we can do to make the site display well in a mobile browser.

**Comment by JamesPierce82** [comment]
Alright. Part of the reason why I asked is because I'm currently in an app development program for school. We're currently taking our third of three android courses, and our second of three iOS courses. I really like the idea behind this and would love to help out if I can, I just don't know how much I can commit to working on stuff right now due to school.

**Comment by drusepth** [comment]
Hey @JamesPierce82. @Cantido's right in that this is something we'd love to implement, but 1) don't currently have the Android dev skills to make it happen, and 2) isn't the highest priority since we're making sure the web experience gets knocked out first. 

However, if you'd be interested in helping out during your time at school, I'd be more than happy to get you set up on our Slack and help out where I can server-side with APIs and such. We're all volunteers here working in whatever free time we have, so that's not an issue. If you'd be interested in starting up a repo for an Android app, just let us know and I can help get the ball rolling. 👍

**Comment by JamesPierce82** [comment]
I'm definitely interested in helping out if I can. What information do you need to get me connected to your slack?

**Comment by mpigsley** [comment]
Hey all, what're your thoughts on using something like react native or ionic to support both iOS and Android together? Might be worth it to go down that route so we don't have to another platform worth of work to do in the future.

My suggestion would probably be to use react native. We could even build it into a [macos app](https://github.com/ptmt/react-native-macos) and a [windows/xbox app](https://github.com/ReactWindows/react-native-windows) all from the same codebase.

**Comment by drusepth** [comment]
@JamesPierce82 I just set up a self-signup for Slack at https://indentlabs.herokuapp.com/ if you want to pop over there -- it'll have you put in your email and send you a Slack invite. FYI most of the fun stuff is in channel #development. (P.S. @mpigsley I don't believe you're on Slack either, so feel free to pop over :+1: )

As for React Native or Ionic, I actually don't know too much about the tech landscape for mobile apps, but this sounds like a good idea in theory. One of the biggest reasons we aren't pursuing mobile apps as a priority-number-one feature right now is that it multiplies the work required to release any future features (i.e. adding a feature for web/ios/android instead of just web), so anything that can be done to share resources or code between multiple platforms sounds like something important to pursue.

**Comment by mpigsley** [comment]
Agreed! With react native it's like building a react app for the web. It's easy for devs proficient in javascript to make the jump. Anyway. My two cents on that.

**Comment by JamesPierce82** [comment]
@mpigsley I like the idea of react, but I tend to find that the experience in apps especially on mobile is much better when built specifically for each specific OS. Obviously react makes things a lot faster to develop for the range, so it's a good option as well.

Definitely worth looking into as keeping things as straight forward as possible for now is probably the best bet. 

I'll sign up when I get a chance, not sure how helpful I'd be working with React as I don't know it, but I have experience in swift, java, JavaScript and PHP amongst a few other dabblings, I imagine it shouldn't be too difficult to pick up.

**Comment by Cantido** [comment]
If we are working in React, I feel like that would just be duplicating work
since Indent is already a web application. I'd rather put development
effort into improving the mobile experience on the web site, and then we
could just make a native app containing a web view or something. Or go the
GitHub route and just have a really good mobile design and not have an app.

Not to say I wouldn't love writing an Android app! But I do have a great
distaste for JavaScript lol.

**Comment by mpigsley** [comment]
@JamesPierce82 react-native compiles to native script.

**Comment by JamesPierce82** [comment]
@mpigsley I know, I more so meant that I've found  coding apps specifically for the native script turns out better than the generically compiled then converted apps. Either way, I'm game to help if I can.

**Comment by mpigsley** [comment]
@Cantido I've added my email to that slack signup but I don't receive an email in response.

**Comment by drusepth** [comment]
@mpigsley That's odd. Shoot me over a quick email to andrew@indentlabs.com and I'll make sure you get your invite. @JamesPierce82 You can do the same if you'd like. Sorry about that, I'll look into what's broken.

**Comment by drusepth** [comment]
Progress on this has started; refer to https://github.com/indentlabs/notebook-android

**Comment by JamesPierce82** [comment]
Yes, Android app in progress. If you want to assign me to this issue @drusepth that'd be fine with me

**Comment by drusepth** [comment]
Got more mentions of this from the feature request form:

>A phone app would be amazing.

(will edit this comment as I see more)

**Comment by JamesPierce82** [comment]
Glad to hear interest in one! I've been focusing on finishing out the final project for this semester, I will have a lot more spare time to work on coding things not school related once it's over.
